test/fx/test_cse_pass.py

```diff
@@ -0,0 +1,240 @@
+# Owner(s): ["oncall: fx"]
+
+import torch
+import torch.fx as fx
+
+from torch.testing._internal.common_utils import (
+    TestCase, run_tests)
+from torch.fx.experimental.proxy_tensor import make_fx
+from torch.fx.passes.dialect.common.cse_pass import CSEPass, get_CSE_banned_ops
+
+import random
+
+
+banned_ops = get_CSE_banned_ops()
+P_default = CSEPass(banned_ops=banned_ops)
+
+def check(f, t, delta, check_val=True, graph_input=False, P=None):
+    """
+    check if the CSE modified graph of ``f``
+    1) has delta less nodes, and
+    2) do not reduce the number of nodes further on a second pass, and
+    3) modified returned is true only if the number of nodes decreases.
+
+    Args:
+        f: function to be checked
+        t: tensor to be passed to f
+        delta: an integer >= -1.
+               If delta = -1, it only checks if the new graph has less or equal number of nodes
+        check_val: if True, check if the output of f is correct
+        graph_input: True is f is type GraphModule
+        P: the pass to use. If None, use P_default
+    """
+    if graph_input:
+        fx_g = f
+    else:
+        fx_g = make_fx(f)(t)
+
+    if P is None:
+        P = P_default
+
+    res = P(fx_g)
+    new_g = res.graph_module
+    new_graph = new_g.graph
+    modified = res.modified
+
+    # the number of nodes decrease/ or stay the same
+    old_num_nodes = len(fx_g.graph.nodes)
+    new_num_nodes = len(new_graph.nodes)
+
+    assert (new_num_nodes < old_num_nodes) == modified, "modified should be True if the number of nodes decrease"
+
+    if delta == -1:
+        assert old_num_nodes >= new_num_nodes, (
+            f"number of nodes increased {old_num_nodes}, {new_num_nodes}")
+    else:
+        assert old_num_nodes == new_num_nodes + delta, (
+            f"number of nodes not the same {old_num_nodes - delta}, {new_num_nodes}\n {fx_g.graph} \n {new_graph}")
+
+    # a second pass should not reduce more nodes
+    res = P(new_g)
+    pass_2_graph = res.graph_module.graph
+    pass_2_num_nodes = len(pass_2_graph.nodes)
+    assert pass_2_num_nodes == new_num_nodes, (
+        f"second pass graph has less node {pass_2_num_nodes}, {new_num_nodes}\n {new_graph} \n {pass_2_graph}")
+
+    # check correctness
+    if check_val:
+        true_result = fx_g(t)
+        our_result = new_g(t)
+        if true_result is None:  # both return None
+            assert our_result is None, f"true result is None, CSE result is {our_result}"
+        else:  # results returned are the same
+            assert torch.all(true_result == our_result), (
```

Comment:
Could you replace the `assert`s with `self.assertTrue(...)` by passing in the `self` from the test functions?